### ISSUE TITLE
[BUG] Fine-tuning cp4 generates strange Training Loss

### ISSUE THREAD

**Issue by Jiadalee** [MouseLand/cellpose#1217]
**Describe the bug**
I'm using the following codes and my own labels to fine-tune the cp4 model, but the train loss starts with a high value, then going down, but finally rise up a lot (see loss figure attached). Such a loss fluctuation pattern persist. Did I set any hyper-parameters incorrectly?

**To Reproduce**
My training and validation codes below:
```
try:
    model_pretrained = models.CellposeModel(gpu=True)
except Exception as e:
    print(f"An error occurred while loading the model: {e}")

#cp4: fine-tune cpsam model (Note: cpsam model only accepts image shape of [batch, height, width] - needs to extract channel)
model_name = "cpsam_finetuned_new"
new_model_path, train_losses, val_losses = train.train_seg(model_pretrained.net,
                              train_data=train_images_copy,
                              train_labels=train_masks_copy,
                              test_data=val_images_copy,
                              test_labels=val_masks_copy,
                              save_path=model_dir,
                              n_epochs=30,
                              learning_rate=0.01,
                              weight_decay=0.0001,
                              SGD=True,                  # set False to Use Adam optimizer
                              nimg_per_epoch=8,
                              model_name=model_name,
                              min_train_masks=0)         # Allow training with fewer masks
```

**Run log**

2025-06-02 09:45:49,614 [INFO] 
cellpose version: 	4.0.5.dev14+gc68d52e 
platform:       	linux 
python version: 	3.12.9 
torch version:  	2.6.0+cu124
2025-06-02 09:45:49,616 [WARNING] SGD is deprecated, using AdamW instead
2025-06-02 09:45:49,617 [INFO] computing flows for labels
100%|████████████████████████████████████████████████████████████████████████████████████| 99/99 [00:03<00:00, 29.05it/s]
2025-06-02 09:45:53,039 [INFO] computing flows for labels

100%|████████████████████████████████████████████████████████████████████████████████████| 11/11 [00:00<00:00, 28.45it/s]
2025-06-02 09:45:53,429 [INFO] >>> computing diameters

100%|██████████████████████████████████████████████████████████████████████████████████| 99/99 [00:00<00:00, 6176.35it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 11/11 [00:00<00:00, 4963.14it/s]
2025-06-02 09:45:53,452 [INFO] >>> normalizing {'lowhigh': None, 'percentile': None, 'normalize': True, 'norm3D': True, 'sharpen_radius': 0, 'smooth_radius': 0, 'tile_norm_blocksize': 0, 'tile_norm_smooth3D': 1, 'invert': False}

2025-06-02 09:45:53,578 [INFO] >>> n_epochs=30, n_train=99, n_test=11
2025-06-02 09:45:53,579 [INFO] >>> AdamW, learning_rate=0.01000, weight_decay=0.00010
2025-06-02 09:45:53,581 [INFO] >>> saving model to Microscopy/Image Analysis/trained_models/models/cpsam_finetuned_new
2025-06-02 09:45:55,596 [INFO] 0, train_loss=0.0417, test_loss=0.0458, LR=0.000000, time 2.02s
2025-06-02 09:46:01,551 [INFO] 5, train_loss=0.1879, test_loss=0.0986, LR=0.005556, time 7.97s
2025-06-02 09:46:09,156 [INFO] 10, train_loss=0.1079, test_loss=0.0946, LR=0.010000, time 15.58s
2025-06-02 09:46:20,561 [INFO] 20, train_loss=0.1116, test_loss=0.0943, LR=0.010000, time 26.98s
2025-06-02 09:46:30,351 [INFO] saving network parameters to Microscopy/Image Analysis/trained_models/models/cpsam_finetuned_new
Diameter of labels in training images: 8.624480247497559

**Screenshots**

![Image](https://github.com/user-attachments/assets/29e89957-d9b8-4dd6-a943-6ac9a72c9f8f)

**Comment by mrariden** [comment]
We only calculate the `test_loss` at [every 10 epochs plus the first epoch](https://github.com/MouseLand/cellpose/blob/main/cellpose/train.py#L475). Otherwise the `test_losses` are padded with zero. We skip this calculation for speed and it's somewhat redundant.
